File: content/problems/2178-maximum-split-of-positive-even-integers/solutions/ascending-even-greedy/solution.cpp
class Solution {
    public:vector<long long> maximumEvenSplit(long long finalSum){
        vector<long long> answer;
        if(finalSum%2)return answer;
        for(long long next=2;finalSum>=next;next+=2){
            answer.push_back(next);
            finalSum-=next;
        }
        if(!answer.empty())answer.back()+=finalSum;
        return answer;
    }
};

File: content/problems/2178-maximum-split-of-positive-even-integers/solutions/ascending-even-greedy/Solution.java
class Solution {
    public List<Long> maximumEvenSplit(long finalSum){
        List<Long> answer=new ArrayList<>();
        if((finalSum&1)==1)return answer;
        for(long next=2;finalSum>=next;next+=2){
            answer.add(next);
            finalSum-=next;
        }
        if(!answer.isEmpty())answer.set(answer.size()-1,answer.get(answer.size()-1)+finalSum);
        return answer;
    }
}
